mirdata/datasets/egfxset.py

```diff
@@ -0,0 +1,388 @@
+"""EGFxSet Dataset Loader
+
+.. admonition:: Dataset Info
+    :class: dropdown
+
+    EGFxSet (Electric Guitar Effects dataset) features recordings for all clean tones in a 22-fret Stratocaster,
+    recorded with 5 different pickup configurations, also processed through 12 popular guitar effects.
+    Our dataset was recorded in real hardware, making it relevant for music information retrieval tasks on real music.
+    We also include annotations for parameter settings of the effects we used.
+
+    EGFxSet is a dataset of 8,970 audio files with a 5-second duration each,
+    summing a total time of - 12 hours and 28 minutes -.
+
+    All possible 138 notes of a standard tuning 22 frets guitar were recorded in each one of the 5 pickup configurations,
+    giving a total of 690 clean tone audio files ( 58 min ).
+
+    The 690 clean audio (58 min) files were processed through 12 different audio effects employing actual guitar gear (no VST emulations were used),
+    summing a total of 8,280 processed audio files (11 hours 30 min).
+
+    The effects employed were divided into four categories, and each category comprised three different effects.
+    Sometimes there were employed more than one effect from a same guitar equipment.
+
+    Categories, Models and Effects:
+
+        Distortion:
+            Boss BD-2: Blues Driver
+            Ibanez Minitube Screamer: Tube Screamer
+            ProCo RAT2: Distortion
+
+        Modulation:
+            Boss CE-3: Chorus
+            MXR Phase 45: Phaser
+            Mooer E-Lady: Flanger
+
+        Delays:
+            Line6 DL-4:
+                        Digital Delay
+                        Tape Echo
+                        Sweep Echo
+
+        Reverb:
+            Orange CR-60 Combo Amplifier:
+                                        Plate Reverb
+                                        Hall Reverb
+                                        Spring Reverb
+
+
+
+    Annotations are labeled by a trained electric guitar musician. For each tone, we provide:
+
+    - Guitar string number
+    - Fret number
+    - Guitar pickup configuration
+    - Effect name
+    - Effect type
+    - Hardware modes
+    - Knob names
+    - Knob types
+    - Knob settings
+
```

Comment:
Nit: you might want to add extra spaces for formating here, see the rendering of the docs [here](https://mirdata--556.org.readthedocs.build/en/556/source/mirdata.html#module-mirdata.datasets.egfxset)

Reply:
I think the Dataset info is more legible now, I've done the changes. In the end, it has two cached properties: note_name and midinote, the first one using a function of annotations and the other one, the class you told us about before. both properties have asserts to see if expected results get accomplished. Please, let me know if more changes are needed. Thanks for all!